sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryOptions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using Azure.Monitor.OpenTelemetry.Exporter;
+
+namespace Azure.Monitor.OpenTelemetry
+{
+    /// <summary>
+    /// Options that allow users to configure the Azure Monitor OpenTelemetry.
+    /// </summary>
+    public class AzureMonitorOpenTelemetryOptions : AzureMonitorExporterOptions
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether Traces should be enabled.
+        /// </summary>
+        public bool EnableTraces { get; set; } = true;
```

Comment:
_ApplicationInsights SDK only has a single DisableTelemetry option as best I can tel_

Application Insights has on/off for each auto-collection module.

Reply:
This is something to be discussed more. I think most users should be good with the defaults. But then they can ask for additional things like  - add custom meter, activitysource, other things.. We should not say "sorry not possible. you have to get rid of distro, and do everything by hand". I am hoping we can say "you can customize otel completely while retaining the distro's one-line, by registering your own Action<providerBuider> to DI".. 

For example: 
AddAzMon()  - adds 4 instrumentaion libraries.
User want to add a 5th one.
it should be something like

AddAzmon() - no change here.
ConfigureTracerProviderBuilder(builder.AddMyNewInstrumentation());

^ Is this your expectation too?

Reply:
I tested it earlier, customer would be able to get additional instrumentations, custom ActivitySource, etc., Syntax is something like you explained ^.

Reply:
I feel like this still leaves several questions...

1. Will users figure that plan out?
2. Will it be a common occurrence that people want to add a little extra, and when they do they will have this (IMO) odd looking configuration code?
3. Assuming they do figure it out and people start mixing and matching the different configuration APIs, how likely is it that a new user seeing this code understands what output it produces?

```C#
services.AddAzureMonitorOpenTelemetry(o =>
{
    o.EnableMetrics = false;
}
services.ConfigureMeterProviderBuilder(b => b.AddMeter("Foo"));
```

If I ignore what I've seen of the implementation and just guess what behavior this should have I don't know what to expect. 


It is good to know what works but I also want to be a little pointed that my concern here is about does the API look nice, will it feel easy to use and be easily understandable. I am thinking of this API as the front door of Azure Monitor for .NET developers for the next decade. It will show up in samples everywhere, it will be first (and perhaps only) lines of code they write, and it will set their first impression.

> Application Insights has on/off for each auto-collection module.

Can you point me at them? I didn't find them either by searching the web or looking at some of the source. Its likely I didn't guess the right terms to search for.

Reply:
> Can you point me at them? 

https://learn.microsoft.com/en-us/azure/azure-monitor/app/asp-net-core?tabs=netcorenew%2Cnetcore6#use-applicationinsightsserviceoptions

^ Most of them were over time due to user demand.

Reply:
> Will users figure that plan out?

I do not see any reason why users have trouble figuring this out... This has been the pattern in ApplicationInsights  (does not mean its a perfect solution! It certainly has a lot of ugliness as it was built over time, and had to keep back-compat, etc, but overall it feels clean IMHO.).

The overall flow is like:
1. Basic usage with all defaults. (used in hello world, pretty much the face of app insights)
`services.AddApplicationInsightsTelemetry();`
 - enables app insights with all defaults (auto collection modules, few initializers, sampling processor, metric extraction, livemetrics, etc.), and reads ConnectionString from IConfig.

2. User want turn on/off things - They are given knobs for most things (on/off for most common feature, *not* all).
3. User want to add more things like more processor, initializers - They just add it to DI, and it'll be picked up.
4. User want to customize each auto-collection module - `ConfigureTelemetryModule<DependencyTrackingTelemetryModule>` with an action to fully customize the auto-collection modules.

The doc for ref: https://learn.microsoft.com/en-us/azure/azure-monitor/app/asp-net-core?tabs=netcorenew%2Cnetcore6#configure-the-application-insights-sdk

Reply:
> If I ignore what I've seen of the implementation and just guess what behavior this should have I don't know what to expect.

Thats fair. I can see it can be hard to predict the behavior without reading accompanying doc

Reply:
In my view, these public API offer customers a quick and easy way to collect telemetry and send data to application insights. We could document limited customization options, such as adding new instrumentation or changing sampling percentages, etc., However, when complex configuration updates are needed for most instrumentations within the TracerProvider/MeterProvider, we should suggest that customers use the "piece meal" approach. This approach involves manually building the TracerProvider/MeterProvider with Azure Monitor Exporter using the documentation provided by OpenTelemetry .NET SDK, without utilizing any of the distro APIs. 

The plan is same for .NET Worker Service, to start with we will not have any package and customer need to take a `piece meal` approach.